fcrepo-clients/src/main/java/edu/unc/lib/dl/fedora/ManagementClient.java

```diff
@@ -258,16 +258,28 @@ public String addInlineXMLDatastream(PID pid, String dsid, boolean force, String
 		return id;
 	}
 
-	public boolean addLiteralStatement(PID pid, String relationship, String literal, String datatype)
+	public void addLiteralStatement(PID pid, String pred, Namespace ns, String literal, String datatype)
 			throws FedoraException {
-		AddRelationship req = new AddRelationship();
-		req.setPid(pid.getURI());
-		req.setDatatype(datatype);
-		req.setIsLiteral(true);
-		req.setRelationship(relationship);
-		req.setObject(literal);
-		AddRelationshipResponse resp = (AddRelationshipResponse) this.callService(req, Action.addRelationship);
-		return resp.isAdded();
+		addTriple(pid, pred, ns, true, literal, datatype);
+	}
+	
+	public void addTriple(PID pid, String pred, Namespace ns, boolean isLiteral, String value, String datatype)
+			throws FedoraException {
+		
+		DatastreamDocument dsDoc = getRELSEXTWithRetries(pid);
```

Comment:
Shouldn't this be inside the loop? If we get an OptimisticLockException, when do we retrieve a new version of the document to work with?